### PR TITLE
Add parent_team_id support

### PR DESCRIPTION
**Pull request by alexnorell**
I noticed that `Organization.create_team()` and `Team.edit()` didn't have the ability to set the parent team ID.

I haven't added any tests, but I can if wanted.

### CODE REVIEW

**Comment by s-t-e-v-e-n-k** [comment]
It looks good, but yes, it will require tests.

**Comment by velicuvlad** [comment]
Is this a feature that will come soon?

**Comment by alexnorell** [comment]
@s-t-e-v-e-n-k, I'm trying to figure out how to generate the ReplayData for the new/edited tests. It looks like `BeaverSoftware` isn't created using the tests. Did I miss a step in `CONTRIBUTING.md`?

```
$ python3 -m pytest -k testCreateTeamWithParent --record 
===================================================================== test session starts =====================================================================
platform darwin -- Python 3.6.8, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /Users/alex/pygithub, configfile: pytest.ini
plugins: cov-2.10.1
collected 666 items / 665 deselected / 1 selected                                                                                                             

tests/Organization.py F                                                                                                                                 [100%]

========================================================================== FAILURES ===========================================================================
____________________________________________________________ Organization.testCreateTeamWithParent ____________________________________________________________

self = <tests.Organization.Organization testMethod=testCreateTeamWithParent>

    def setUp(self):
        super().setUp()
>       self.org = self.g.get_organization("BeaverSoftware")

tests/Organization.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
github/MainClass.py:299: in get_organization
    headers, data = self.__requester.requestJsonAndCheck("GET", "/orgs/" + login)
github/Requester.py:319: in requestJsonAndCheck
    verb, url, parameters, headers, input, self.__customConnection(url)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <github.Requester.Requester object at 0x7ff91835b6d8>, status = 404
responseHeaders = {'DEBUG_FRAME': 0, 'access-control-allow-origin': '*', 'access-control-expose-headers': 'ETag, Link, Location, Retry-A..., X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset', 'content-encoding': 'gzip', ...}
output = {'documentation_url': 'https://docs.github.com/rest/reference/orgs#get-an-organization', 'message': 'Not Found'}

    def __check(self, status, responseHeaders, output):
        output = self.__structuredFromJson(output)
        if status >= 400:
>           raise self.__createException(status, responseHeaders, output)
E           github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/orgs#get-an-organization"}

github/Requester.py:342: UnknownObjectException
=================================================================== short test summary info ===================================================================
FAILED tests/Organization.py::Organization::testCreateTeamWithParent - github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "document...
============================================================== 1 failed, 665 deselected in 1.32s ==============================================================

**Comment by AlexandreODelisle** [comment]
@alexnorell, it seems you are trying to re-use the  tests/Organization.py, but at the start of the Unitest there is the setUp that is always executed, and you do not have access to the BeaverSoftware Organization.. Hopes it help

File: github/Organization.py [hunk 1]
```diff
@@ -599,6 +600,7 @@ def create_team(
         :param permission: string
         :param privacy: string
         :param description: string
+        :param parent_team_id: string
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:603]: This states parent_team_id is string, but the typing hint says int.

File: github/Team.py [hunk 2]
```diff
@@ -280,13 +280,15 @@ def edit(
...
         privacy=github.GithubObject.NotSet,
+        parent_team_id=github.GithubObject.NotSet,
     ):
         """
         :calls: `PATCH /teams/:id <http://developer.github.com/v3/orgs/teams>`_
         :param name: string
         :param description: string
         :param permission: string
         :param privacy: string
+        :param parent_team_id: string
```
Comment (s-t-e-v-e-n-k) [inline github/Team.py:291]: As above.

File: tests/Organization.py [hunk 3]
```diff
@@ -162,6 +162,11 @@ def testCreateTeamWithAllArguments(self):
         self.assertEqual(team.id, 189852)
         self.assertEqual(team.description, "Description also created by PyGithub")
 
+    def testCreateTeamWithParent(self):
```
Comment (s-t-e-v-e-n-k) [inline tests/Organization.py:165]: There is a replay data file that is missing here, which is why your test is failing. Did you forget to `git add` it, or have you not generated it?

File: tests/Team.py [hunk 4]
```diff
@@ -144,16 +144,19 @@ def testEditWithoutArguments(self):
         self.assertEqual(self.team.name, "Name edited by PyGithub")
 
     def testEditWithAllArguments(self):
+        parent_team = self.org.create_team(name="parent_team")
```
Comment (s-t-e-v-e-n-k) [inline tests/Team.py:147]: This will also require changes to the replay data file for this test too.

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
This is looking like a good start, just needs the tests sorted out completely and some docstring changes.

File: tests/Organization.py (hunk 3 above)
Comment (alexnorell) [reply tests/Organization.py:165]: I was having issues generating it. https://github.com/PyGithub/PyGithub/pull/1669#issuecomment-695116291

File: github/Organization.py (hunk 1 above)
Comment (alexnorell) [reply github/Organization.py]: Updated with https://github.com/PyGithub/PyGithub/pull/1669/commits/2c272f8b58e14cfbde843f07967a7003a62dae78

File: github/Team.py (hunk 2 above)
Comment (alexnorell) [reply github/Team.py]: Updated with https://github.com/PyGithub/PyGithub/pull/1669/commits/2c272f8b58e14cfbde843f07967a7003a62dae78

File: tests/Organization.py (hunk 3 above)
Comment (s-t-e-v-e-n-k) [reply tests/Organization.py:165]: Some of the older tests include references to PRs/repos and so on that don't exist any more. :-( In this case, you can either comment out setUp so doesn't try and fetch it and then fail (and then revert that once your replay data is written), or separate it out to a different test file.

**Comment by phspagiari** [comment]
This will be available soon? Thanks

**Pull request closed without merging**